### PR TITLE
[WIP] refactor(unstable): change return types in runtime compiler APIs

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes https://github.com/denoland/deno/issues/4752

CC @kitsonk

### CODE REVIEW

File: cli/dts/lib.deno.unstable.d.ts [hunk 1]
```diff
@@ -524,7 +530,10 @@ declare namespace Deno {
     rootName: string,
     sources?: Record<string, string>,
     options?: CompilerOptions,
```
Comment (bartlomieju) [inline cli/dts/lib.deno.unstable.d.ts:532]: Removing `Deno.transpileOnly()` in favor of `noCheck` option means we need to change `options?: CompilerOptions` into:
```
options?: {
  noCheck?: boolean,
  compilerOptions?: CompilerOptions
}
```

File: cli/dts/lib.deno.unstable.d.ts [hunk 2]
```diff
@@ -524,7 +530,10 @@ declare namespace Deno {
...
-  ): Promise<[Diagnostic[] | undefined, Record<string, string>]>;
+  ): Promise<{
+      diagnostics: Diagnostic[] | undefined, 
+      emitMap: Record<string, string>
```
Comment (bartlomieju) [inline cli/dts/lib.deno.unstable.d.ts:535]: rename to `emittedFiles`

File: cli/dts/lib.deno.unstable.d.ts [hunk 3]
```diff
@@ -524,7 +530,10 @@ declare namespace Deno {
     rootName: string,
     sources?: Record<string, string>,
     options?: CompilerOptions,
-  ): Promise<[Diagnostic[] | undefined, Record<string, string>]>;
+  ): Promise<{
+      diagnostics: Diagnostic[] | undefined,
```
Comment (bartlomieju) [inline cli/dts/lib.deno.unstable.d.ts:534]: add `ignoredCompilerOptions` field

File: cli/dts/lib.deno.unstable.d.ts (hunk 1 above)
Comment (kitsonk) [reply cli/dts/lib.deno.unstable.d.ts:532]: I suspect there are other non-tsconfig related options that we will want to make in here as well, like import map, so yeah, let's change it.

**Comment by bartlomieju** [comment]
Waiting to close https://github.com/denoland/deno/issues/7225 before finishing this PR

**Comment by bartlomieju** [comment]
Stale, I'll open a new PR

**Pull request closed without merging**